components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/ciba/CibaAuthResponseHandler.java

```diff
@@ -0,0 +1,143 @@
+     * @return Response for AuthenticationRequest.
+     */
+    public Response createAuthResponse(@Context HttpServletResponse response,
+                                       CibaAuthCodeResponse cibaAuthCodeResponse) {
+
+        // Set the ExpiryTime.
+        long expiresIn = cibaAuthCodeResponse.getExpiresIn();
+        if (log.isDebugEnabled()) {
+            log.info("Setting ExpiryTime for the response to the  request made by client with clientID : " +
+                    cibaAuthCodeResponse.getClientId() + ".");
+        }
+        // Create authentication response.
+        response.setContentType(MediaType.APPLICATION_JSON);
+
+        // Creating authentication response for the request.
+        JSONObject cibaAuthResponse = new JSONObject();
+        cibaAuthResponse.put(CibaConstants.AUTH_REQ_ID, cibaAuthCodeResponse.getAuthReqId());
+        cibaAuthResponse.put(CibaConstants.EXPIRES_IN, expiresIn);
+        cibaAuthResponse.put(CibaConstants.INTERVAL, CibaConstants.INTERVAL_DEFAULT_VALUE_IN_SEC);
+
+        if (log.isDebugEnabled()) {
+            log.info("Creating CIBA Authentication response to the request made by client with clientID : " +
+                    cibaAuthCodeResponse.getClientId() + ".");
+        }
+        Response.ResponseBuilder respBuilder = Response.status(HttpServletResponse.SC_OK);
+        if (log.isDebugEnabled()) {
+            log.info("Returning CIBA Authentication Response for the request made by client with clientID : " +
+                    cibaAuthCodeResponse.getClientId() + ".");
+        }
+        return respBuilder.entity(cibaAuthResponse.toString()).build();
+    }
+
+    /**
+     * Creates CIBA Authentication Error Response.
+     *
+     * @param cibaAuthFailureException Ciba Authentication Failed Exception.
+     * @return response Authentication Error Responses for AuthenticationRequest.
+     */
+    public Response createErrorResponse(CibaAuthFailureException cibaAuthFailureException) {
+
+        // Create CIBA Authentication Error Response.
+        if (log.isDebugEnabled()) {
+            log.debug("Creating Error Response for CIBA Authentication Request.");
+        }
+
+        if (cibaAuthFailureException.getErrorCode().equals(OAuth2ErrorCodes.SERVER_ERROR)) {
+
+            return handleServerError(cibaAuthFailureException);
+        } else {
+            return handleClientException(cibaAuthFailureException);
+        }
+    }
+
+    /**
+     * Handles client exception.
+     *
+     * @param cibaAuthFailureException Authentication Failure Exception.
+     * @return Response for AuthenticationRequest.
+     */
+    private Response handleClientException(CibaAuthFailureException cibaAuthFailureException) {
+
+        String errorCode = cibaAuthFailureException.getErrorCode();
+        JSONObject cibaErrorResponse = new JSONObject();
+        cibaErrorResponse.put("error", cibaAuthFailureException.getErrorCode());
```

Comment:
can we use constant for error and error_description

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1701